src/models.py

```
@@ -4131,12 +4131,14 @@ def get_main_strain(self, type=None):
         main_strain = None
         for strain in main_strain_list:
             x = DBSession.query(Straindbentity).filter_by(display_name=strain, subclass='STRAIN').one_or_none()
-            y = DBSession.query(Dnasequenceannotation).filter_by(taxonomy_id=x.taxonomy_id, dbentity_id=self.dbentity_id, dna_type='GENOMIC').one_or_none()
-            if y is None:
+            y = DBSession.query(Dnasequenceannotation).filter_by(taxonomy_id=x.taxonomy_id, dbentity_id=self.dbentity_id, dna_type='GENOMIC').all() 
+            if len(y) == 0:
```

This will fail if y is None.
`if y:` can cover both cases when array is empty or when None object is returned from the query

---

Good catch! but I didn't know fetch all will return None. If nothing found, it will just return an empty list right?

---

If something is changed, for example, table/column names got changed, we do want it to throw error instead of just skipping the check. Right?

---

> Good catch! but I didn't know fetch all will return None. If nothing found, it will just return an empty list right?

That's true. ``all()`` will return empty array if no data found. I was thinking more for an edge case which may not occur but if it does it will throw an exception. Should have left it as suggestion instead of requesting a change :) so up to you.

---

> If something is changed, for example, table/column names got changed, we do want it to throw error instead of just skipping the check. Right?

Yes, try-catch would be nice but also optional depending how you want to go around it.